src/main.rs

```diff
@@ -70,33 +70,36 @@ async fn create_global_svc(
     labels.insert("linkerd.io/service-mirror".to_string(), "true".to_string());
     labels.insert("linkerd.io/global-mirror".to_string(), "true".to_string());
 
-    let svc_meta: ObjectMeta = ObjectMeta {
+    let svc_meta = ObjectMeta {
         labels: Some(labels.clone()),
         name: Some(format!("{svc_name}-svc-global")),
         namespace: Some(ns.to_string()),
         ..ObjectMeta::default()
     };
 
-    let svc_spec: ServiceSpec = ServiceSpec {
+    let svc_spec = ServiceSpec {
         cluster_ip: Some("None".to_string()),
         ports: Some(svc_port),
         selector: Some(labels.clone()),
         ..Default::default()
     };
 
-    let ServiceToCreate: Service = Service {
+    let service_to_create = Service {
         metadata: svc_meta,
         spec: Some(svc_spec),
         ..Service::default()
     };
 
-    //create service
-    let svc: Api<Service> = Api::namespaced(ctx.client.clone(), &ns);
+    // create service
+    let svc: Api<Service> = Api::namespaced(ctx.0.clone(), &ns);
 
-    match svc.create(&PostParams::default(), &ServiceToCreate).await {
+    // Why handle this here instead of bubbling up?
+    match svc.create(&PostParams::default(), &service_to_create).await {
```

Comment:
This could have been greatly simplified by letting the caller handle the error. As a matter of fact, the code you have in here doesn't actually return an error at any point, it always returns Ok.